Sources/Madness/DBConnectionManager/KueryDBConnectionManager.swift

```
@@ -0,0 +1,69 @@
+//
+//  KueryDBConnectionManager.swift
+//  Madness
+//
+//  Created by Ryan Krol on 29/07/2018.
+//
+//  Song: Portugal. The Man - Noise Pollution
+
+import Foundation
+import SwiftKuery
+
+public protocol KueryDBConnectionManager: DBConnectionManager {
+
+}
+
+extension KueryDBConnectionManager {
+
+    /**
+     Will be responsible for handling the work around setting up a database connection,
+     and sending off requests, as well as error handling. Working with the response will
+     happen in the responseHandler
+     */
+    func executeDatabaseAction(
+        parameters: [Any?],
+        query: Query,
+        defaultResponse: SimpleResponse,
+        dbConnectionPool: ConnectionPool,
+        responseHandler: @escaping (
+            _ rows: [[String:Any?]]
+        ) -> SimpleResponse
+    ) -> SimpleResponse {
+
+        var response = defaultResponse
+        var transactionError: Error?          = nil
+        var positiveResponse: SimpleResponse? = nil
+
+        if let connection = dbConnectionPool.getConnection() {
+            connection.execute(
+                query: query,
+                parameters: parameters
+            ) { (queryResult) in
+
+                if let error = queryResult.asError {
+                    transactionError = error
+                }
+
+                if let rows = queryResult.asRows {
+                    positiveResponse = responseHandler(rows)
+                }
+
+            }
+        }
+
+        if let error = transactionError {
+            response = SimpleResponse(
+                message: error.localizedDescription,
+                error: true,
+                restResponseCode: .internalServerError
+            )
```

It would be better to throw an error

---

Talk over lunch